### DISCUSSION TITLE
bad-precaching-response

### DISCUSSION THREAD

**Discussion by VinceBT-BG** [serwist/serwist#228] [Q&A]
With the base config I am getting this using Next 15 with a basic architecture, I have tried excluding the file in question but it does nothing. Is there a known issue and can I do something ?
```
PrecacheStrategy.js:150 Uncaught (in promise) bad-precaching-response: bad-precaching-response :: [{"url":"https://app.example.com/_next/dynamic-css-manifest.json","status":404}]
    at Z.S (https://app.example.com/workbox-d25a3628.js:1:16900)
    at async Z.U (https://app.example.com/workbox-d25a3628.js:1:16452)
    at async Z.q (https://app.example.com/workbox-d25a3628.js:1:7597)
```

My next.config.ts part about the SW is like this:
```
const withSerwist = withSerwistInit({
	swSrc: 'service-worker/index.ts',
	swDest: 'public/sw.js',
})
```

My service worker is basically this:
```
import { defaultCache } from '@serwist/next/worker'
import { BackgroundSyncQueue, type PrecacheEntry, Serwist, type SerwistGlobalConfig } from 'serwist'

declare global {
	interface WorkerGlobalScope extends SerwistGlobalConfig {
		__SW_MANIFEST: (PrecacheEntry | string)[] | undefined
	}
}

declare const self: ServiceWorkerGlobalScope

const queue = new BackgroundSyncQueue('baseQueue')

const serwist = new Serwist({
	precacheEntries: self.__SW_MANIFEST,
	skipWaiting: true,
	clientsClaim: true,
	runtimeCaching: defaultCache,
})

self.addEventListener('fetch', event => {
	const backgroundSync = async () => {
		try {
			const response = await fetch(event.request.clone())
			return response
		} catch (error) {
			await queue.pushRequest({ request: event.request })
			return Response.error()
		}
	}
	event.respondWith(backgroundSync())
})

serwist.addEventListeners()
```

**Comment by katiasmet-93** [comment]
Downgrading to Next 14 temporary fixes the issue.

**Comment by VinceBT-BG** [comment]
What currently worked for me is this config btw:
```ts
const withSerwist = withSerwistInit({
	swSrc: 'service-worker/index.ts',
	swDest: 'public/sw.js',
	exclude: [
		// https://github.com/shadowwalker/next-pwa/issues/424#issuecomment-1332258575
		({ asset }) => {
			// Add here any file that fails pre-caching
			const excludeList = [
				// Default Serwist https://serwist.pages.dev/docs/next/configuring/exclude
				/\.map$/,
				/^manifest.*\.js$/,
				/^server\//,
				/^(((app-)?build-manifest|react-loadable-manifest|dynamic-css-manifest)\.json)$/,
			]
			if (excludeList.some(r => r.test(asset.name))) {
				return true
			}
			return false
		},
	],
})
```

**Reply by khuezy** [reply]
Hmm this somehow breaks Chrome oauth redirects.
For some reason, only Chrome fails to make the `/api/auth/callback` successfully. The service worker is somehow caching that endpoint, as I see the "initiator" for that request as "preload" in the Network tab.

I had to add this to the top of my matcher:
```
{
	matcher: /api\/auth\/.*/i,
	handler: new NetworkOnly(),
},
```

CC @DuCanhGH , did you ever figure this out: https://github.com/serwist/serwist/blob/main/packages/next/src/index.worker.ts#L190

**Reply by DuCanhGH** [reply]
@khuezy I'm not too sure if this issue is related to the discussion... However, no, I haven't investigated this issue, unfortunately. With `@serwist/next`'s default runtime caching, /api/auth/callback routes would not be cached, while the rest of /api/auth would. I wonder what bug are you facing, and on which /api/auth route? Thanks! I don't see how /api/auth/login would fail though. That's just a static page, right?

**Reply by khuezy** [reply]
Sorry I could move this to another discussion. But it's sort of related b/c this error surfaced when I added the `exclude` stuff above to allow the service worker to properly install.

When I sign in to an oauth provider, it calls the `api/auth/callback`. For some reason, the nextjs or authjs causes the service worker to cache that resource and it immediately throws an error (but on Chrome only)

**Reply by DuCanhGH** [reply]
@khuezy are you sure that /api/auth/callback was cached? I don't see how this is the case with `defaultCache`, since /api/auth/callback would straight up fall through all the defined entries. I'd check the caches in the browser to be sure.

**Reply by khuezy** [reply]
I'm not 100% sure, but the Network tab, it showed the `/api/auth/callback` initiator was `Preload`. I guess Chrome specifically doesn't like that and throws an error.

**Reply by DuCanhGH** [reply]
@khuezy `Preload` requests are actually not initiated by your service worker. Those initiated by it would actually display either `ServiceWorker` or `ServiceWorkerPreload` (something like this, but I don't really use Chromium browsers so). I guess there might be a different issue.

**Reply by khuezy** [reply]
Yea I'm not entirely sure what's going on. When I add the `NetworkOnly` at the top, the problem goes away. I'll probably investigate later when I have the time. Thanks!

**Comment by DuCanhGH** [comment]
...That's an old one. I guess we'll just filter out all the JSON files in the compilation directory so that this doesn't happen again :)
